### PR TITLE
Deprecate sweep synthesis

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

As discussed during the weekly

### Changes proposed in this pull request:

- deprecate `pyfar.signals.linear_sweep` in favor of `pyfar.signals.linear_sweep_time`
- deprecate `pyfar.signals.exponential_sweep` in favor of `pyfar.signals.exponential_sweep_time`
- rename tests and add tests for deprecation

### CODE REVIEW

**Review by sikersten — APPROVED** [review]

File: pyfar/signals/deterministic.py [hunk 1]
```diff
@@ -131,6 +132,20 @@ def impulse(n_samples, delay=0, amplitude=1, sampling_rate=44100):
 
 def linear_sweep(n_samples, frequency_range, n_fade_out=90, amplitude=1,
                  sampling_rate=44100):
+    """
+    This function will be deprecated in pyfar 0.5.0. See
+    :py:func:`linear_sweep_time`.
+    """
+
+    warnings.warn(("This function will be deprecated in pyfar 0.5.0 in favor "
+                   "of pyfar.signals.linear_sweep_time."))
```
[the remark this answers is not in the document]
Comment (mberz) [reply pyfar/signals/deterministic.py:141]: Let's use the respective type here :)

File: pyfar/signals/deterministic.py [hunk 2]
```diff
@@ -181,8 +196,22 @@ def linear_sweep(n_samples, frequency_range, n_fade_out=90, amplitude=1,
...
-                      sweep_rate=None, sampling_rate=44100):
+def exponential_sweep(n_samples, frequency_range, n_fade_out=90,
+                      amplitude=1, sweep_rate=None, sampling_rate=44100):
+    """
+    This function will be deprecated in pyfar 0.5.0. See
+    :py:func:`exponential_sweep_time`.
+    """
+
+    warnings.warn(("This function will be deprecated in pyfar 0.5.0 in favor "
+                   "of pyfar.signals.exponential_sweep_time."))
```
Comment (mberz) [inline pyfar/signals/deterministic.py:207]:
```suggestion
    warnings.warn(("This function will be deprecated in pyfar 0.5.0 in favor "
                   "of pyfar.signals.exponential_sweep_time."), PendingDeprecationWarning)
```

File: tests/test_signals.py [hunk 3]
```diff
@@ -329,9 +342,19 @@ def test_exponential_sweep_rate():
...
 
-def test_exponential_sweep_assertion():
+def test_exponential_sweep_time_assertion():
     with pytest.raises(ValueError, match="The exponential sweep can not"):
-        pfs.exponential_sweep(2**10, [0, 20e3])
+        pfs.exponential_sweep_time(2**10, [0, 20e3])
+
+
+def test_exponential_sweep_deprecation():
+    with pytest.warns(UserWarning, match="This function will be deprecated"):
```
Comment (mberz) [inline tests/test_signals.py:351]:
```suggestion
    with pytest.warns(PendingDeprecationWarning, match="This function will be deprecated"):
```

File: tests/test_signals.py [hunk 4]
```diff
@@ -266,37 +269,47 @@ def test_linear_sweep_against_reference():
...
-        pfs.linear_sweep(100, 1)
+        pfs.linear_sweep_time(100, 1)
     with pytest.raises(ValueError, match="Upper frequency limit"):
-        pfs.linear_sweep(100, [1, 40e3])
+        pfs.linear_sweep_time(100, [1, 40e3])
 
 
-def test_exponential_sweep_against_reference():
+def test_linear_sweep_deprecation():
+    with pytest.warns(UserWarning, match="This function will be deprecated"):
```
Comment (mberz) [inline tests/test_signals.py:299]:
```suggestion
    with pytest.warns(PendingDeprecationWarning, match="This function will be deprecated"):
```

**Review by mberz — CHANGES_REQUESTED** [review]
One minor thing: Let's replace the UserWarning with PendingDeprecationWarning

**Review by mberz — APPROVED** [review]

**Pull request merged**